### PR TITLE
[GA] Add GA4 timing event around detec/fulfill endpoints

### PR DESCRIPTION
**Pull request by beets**
This is still somewhat experimental, since GA4 removed native support for page timing events:
https://support.google.com/analytics/answer/9964640#settings-with-no-GA4-equivalent&zippy=%2Cin-this-article

@pradh Adding minimal information first that we can continue to augment if this works, though please let me know if any of the other parameters would be useful to track.

### CODE REVIEW

File: static/js/apps/explore/app.tsx [hunk 1]
```diff
@@ -330,6 +336,17 @@ const fetchFulfillData = async (
       classifications: classificationsJson,
       disableExploreMore,
     });
+    if (startTime) {
+      const endTime = window.performance
```
Comment (chejennifer) [inline static/js/apps/explore/app.tsx:340]: nit: this wouldn't be end time right? this is the time taken?

File: static/js/shared/ga_events.ts [hunk 2]
```diff
@@ -26,6 +26,7 @@ export function triggerGAEvent(
   if (window.gtag) {
     window.gtag("event", eventName, parameter);
   }
+  console.log(eventName, parameter);
```
Comment (chejennifer) [inline static/js/shared/ga_events.ts:29]: nit: I don't think we need this?

File: static/js/shared/ga_events.ts [hunk 3]
```diff
@@ -97,6 +98,22 @@ export const GA_EVENT_TOOL_CHART_OPTION_CLICK = "tool_chart_option_click";
...
+ */
+export const GA_EVENT_NL_DETECT_FULFILL = "explore_detect_fulfill";
+
+/**
+ * Triggered when detection results are returned in NL search.
+ * Parameters:
+ *   "topic": array of topics
+ *   "time_ms": e2e request/response timing in ms.
+ */
+export const GA_EVENT_NL_FULFILL = "explore_fulfill";
```
Comment (chejennifer) [inline static/js/shared/ga_events.ts:115]: fulfill is based on topic + place, should we also include places as a parameter?

**Review by chejennifer — APPROVED** [review]
nice! Just some nits/suggestions

File: static/js/shared/ga_events.ts (hunk 2 above)
Comment (beets) [reply static/js/shared/ga_events.ts:29]: thanks for catching that!

File: static/js/apps/explore/app.tsx (hunk 1 above)
Comment (beets) [reply static/js/apps/explore/app.tsx:340]: yes, elapsed time - updated.

**Review by pradh — APPROVED** [review]
Query is good.

Past queries in the context would also be useful since that can impact the latency of a query.  But that can come after.

**Comment by beets** [comment]
thanks for the reviews! i've added a todo to add past queries.

**Pull request merged**